_posts/2017-10-17-faf-in-openshift.markdown

```diff
@@ -0,0 +1,160 @@
+2. Add the FAF application. We've created docker image with FAF, that you can find on
+[Docker Hub](https://hub.docker.com/r/abrt/faf-image).
+```
+oc new-app abrt/faf-image --name <app_name>
+```
+This command will pull the image from docker hub and then deploy it.
+
+3. Edit Security Context Constraints. <br><br>
+The deployment won't succeed because the FAF container requires usage of the root user, but default
+configuration doesn't allow it. To allow the container to use of root, create a new service account
+and add it to **anyuid's** security context constraints.
+```
+oc create serviceaccount <sa_username>
+oc adm add-scc-to-user anyuid -z <sa_username>
+oc patch dc/<app_name> --patch '{"spec":{"template":{"spec":{"serviceAccountName": "<sa_username>"}}}}'
+```
+Changing deployment configuration will trigger new container deployment. This time, it should succeed,
+and the FAF should be running and accessible through ```http://<pod_ip>/faf```.
+<br><br>
+To get IP address of the pod:
+```
+{% raw %}oc get pod <pod_name> --template='PodIP: {{.status.podIP}}{{"\n"}}'{% endraw %}
+```
+
+
+## Web Interface
+
+It is also possible to deploy FAF through the web interface, except for editing security context
+constraints. You will need to use the command line for it.
+
+1. Create a new project.
+[![create_project][create_project]][create_project]
+
+2. Add FAF's docker image to project. Use image name ```abrt/faf-image/```.
+[![add_docker_image][add_docker_image]][add_docker_image]
+
+3. Create a new service account and add it to **anyuid's** SCC.
+```
+oc create serviceaccount <sa_username>
+oc adm policy add-scc-to-user anyuid -z <sa_username>
+oc patch dc/<app_name> --patch '{"spec":{"template":{"spec":{"serviceAccountName": "<sa_username>"}}}}'
+```
+The FAF should now be running on ```http://<pod_ip>/faf```.
+[![success_deploy][success_deploy]][success_deploy]
+
+
+# Persistent storage for FAF
+
+Containers are not persistent by default; their contents get cleared, when you restart them.
+Default claimed volume for the pod is EmptyDir, which is a temporary directory that will get cleared, once the pod is terminated.
+Again, you have an option to do this either using the command line or the web interface.
+
+### Command line
+
+Create a volume claim for database:
+```yaml
+apiVersion: "v1"
+kind: "PersistentVolumeClaim"
+metadata:
+  name: "faf-database"
+spec:
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "5Gi"
+```
+```
+oc create -f db-pvc-claim.yaml
+```
+
+Create a volume claim for FAF's data:
+```yaml
+apiVersion: "v1"
+kind: "PersistentVolumeClaim"
+metadata:
+  name: "faf-report-data"
+spec:
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "5Gi"
+```
+```
+oc create -f report-pvc-claim.yaml
+```
+
+Update the volume configuration:
+```
+oc volume dc/<app_name> --add --overwrite --claim-name=faf-database --name=faf-1 -t=persistentVolumeClaim -m=/var/lib/postgres
```

Comment:
oc set volume dc/<app_name> --add --overwrite --name=faf-1 -t=pvc -m=/var/lib/postgres --claim-name=faf-database
 --claim-size=5Gi --claim-mode=ReadWriteOnce

With this you do not need to create the volume claim using 'oc create -f db-pvc-claim.yaml' as it will be created in one step. I just learned this an hour ago :)  The same for the volume command bellow.

Reply:
It doesn't work on my machine. I have to manually create PVC and then add volume to it. I've just tried it. Are there any settings to turn that behavior on?

Reply:
I have PersistentVolume (PV) with Accessmode RWX. And this works for me:
oc set volume dc/<app_name> --add --overwrite --name=faf-volume-1 -t=pvc  --claim-name=faf-database --claim-size=5Gi --claim-mode='ReadWriteMany'
What error are you getting?

Reply:
Your command works! You have to specify the claimed size and mode. I will change it in the article. Thank you.